implementations/rust/ockam/ockam_identity/src/secure_channel/options.rs

```diff
@@ -63,21 +68,30 @@ impl SecureChannelOptions {
         self
     }
 
-    /// Adds provided credentials
-    pub fn with_credentials(mut self, credentials: Vec<CredentialAndPurposeKey>) -> Self {
-        self.credentials.extend(credentials);
-        self
+    /// Set [`CredentialRetriever`]
+    pub fn with_credential_retriever(
+        mut self,
+        credential_retriever: Arc<dyn CredentialRetriever>,
+    ) -> Result<Self> {
+        if self.credential_retriever.is_some() {
+            return Err(IdentityError::CredentialRetrieverAlreadySet.into());
+        }
+        self.credential_retriever = Some(credential_retriever);
+        Ok(self)
     }
 
-    /// Adds a single credential
-    pub fn with_credential(mut self, credential: CredentialAndPurposeKey) -> Self {
-        self.credentials.push(credential);
-        self
+    /// Set credential
+    pub fn with_credential(mut self, credential: CredentialAndPurposeKey) -> Result<Self> {
+        if self.credential_retriever.is_some() {
```

Comment:
This could be directly:
```rust
self.with_credential_retriever(Arc::new(MemoryCredentialRetriever::new(credential)))
```

Reply:
Sure, but it's less convenient IMO. Also requires you to know about `MemoryCredentialRetriever` existence

Reply:
I meant "as the implementation of the `with_credential`" method.

Reply:
Oh, sure